scene/gui/line_edit.cpp

```diff
@@ -369,6 +371,11 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 				selection_fill_at_caret();
 			}
 		}
+
+		if (drag_action && is_editable() && get_viewport()->gui_get_drag_data().get_type() == Variant::STRING) {
```

Comment:
Actually if you have access to the drag data (I didn't actually know it's available like that...), you could just use `can_drop_data(get_viewport()->gui_get_drag_data())`.

So the condition would become
```
if (drag_action && can_drop_data(get_viewport()->gui_get_drag_data()))
```